### ISSUE TITLE
There are serious issues with tab hiding.

### ISSUE THREAD

**Issue by mzso** [NiklasGollenstede/tst-search#5]
Hi!
I noticed serious issue with tab hiding. Apparently all of them are hidden each time I close and open TST. I have to search for something then delete it then press enter again. (Which magnifies how annoying it is the lack of a convenient X in the right of the input to clear it.)
Tabs also disappeared eclectically during use, such as when I moved one to top level. I need to use the same workaround to get the tab back. So far I didn't notice any particular patter for this.

**Comment by mzso** [comment]
On second look. It seems like moving a child to top level always causes it to disappear.
I'm not sure if it's the only event to cause tabs to get hidden erroneously.

**Comment by mzso** [comment]
[**@mzso**](https://github.com/mzso) commented on [2021. ápr. 30. 22:20 CEST](https://github.com/NiklasGollenstede/tst-search/issues/5#issuecomment-830358912 "2021-04-30T20:20:52Z - Replied by Github Reply Comments"):
> On second look. It seems like moving a child to top level always causes it to disappear.
> I'm not sure if it's the only event to cause tabs to get hidden erroneously.

Observing a bit more, it doesn't seem to always glitch. But changing to another addon's sidebar and back TST seems to always get the tabs to disappear.

**Comment by NiklasGollenstede** [comment]
(Ups. Was referencing the wrong issue in the commit there.)

Yes, I have noticed similar stuff before, only that by tabs never completely disappeared (but that is probably a matter of the selected result highlighting options).

I am fairly certain that this is an issue with TST itself. I opened an issue for that: https://github.com/piroor/treestyletab/issues/2882
Under certain circumstances, TST seems to revive removed tab states.

While removing them again helps (and is what happens if you do an empty search), this extension wouldn't know when to do that.
